presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractRowBlock.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public abstract class AbstractRowBlock
+        implements Block
+{
+    protected final int numColumns;
+
+    protected abstract Block getColumnBlock(int columnIndex);
```

Comment:
One difference is that `getBlocks` cannot be executed efficiently with `RowBlockBuilder`: it must do the element-wise cast (BlockBuilder[] -> Block[])